deploy/roles/spark/tasks/main.yml

```diff
@@ -0,0 +1,92 @@
+---
+- name: Create service account for Spark
+  user: 
+    name: "{{ spark_user }}"
+    system: yes
+    home: "{{ spark_lib_dir }}"
+    shell: "{{ spark_user_shell }}"
+    state: present
+    groups: "{{ spark_user_groups | join(',') }}"
+    comment: "Spark-User"
+
+- name: Make sure Spark configuration directory exists
+  file: 
+    path: "{{ spark_conf_dir }}"
+    state: directory
+
+- name: Make sure Spark directories exists
+  file: 
+    path: "{{ item }}" 
+    owner: "{{ spark_user }}"
+    group: "{{ spark_user }}"
+    mode: 0755
+    state: directory
+  with_items:
+    - "{{ spark_run_dir }}"
+    - "{{ spark_main_dir }}"
+
+- name: Ensure Spark log directory with appropriate permissions
+  file: 
+    path: "{{ item }}" 
+    owner: "{{ spark_user }}"
+    group: "{{ spark_user }}"
+    mode: 1755
+    state: directory
+  with_items:
+    - "{{ spark_log_dir }}"
+
+- name: Download Spark
+  get_url: 
+    url: "{{ spark_mirror }}"                                                             
+    dest: "{{ spark_src_dir }}"
+#    validate_certs: no
+
+- name: Extract Spark
+  unarchive: 
+     src: "{{ spark_src_dir }}/{{ spark_tar_file }}"                                                          
+     dest: "{{ spark_usr_parent_dir }}"                                                       
+     copy: no
+
+- name: Spark symlinks
+  file: 
+     src: "{{ item.src }}"
+     dest: "{{ item.dest }}"
+     state: link
+  with_items:
+    - { src: "{{ spark_main_dir }}/conf", dest: "{{ spark_conf_dir }}/conf" }
+
+- name: Spark binaries set up
+  template: 
+    src: spark-bin.j2
+    dest: "/usr/bin/{{ item }}"
+    mode: 0755
+  with_items:
+    - spark-class
+    - spark-shell
+    - spark-sql
+    - spark-submit
+
+- name: Set up SPARK_HOME environment variables
+  template: 
+    src: "spark-env.sh.j2"
+    dest: "{{ spark_main_dir }}/conf/spark-env.sh"
+
+- name: Set up spark defaults config file
+  template: 
+    src: spark-defaults.conf.j2
+    dest: "{{ spark_main_dir }}/conf/spark-defaults.conf"
+
+- name: Set up spark log4j config
+  template: 
+    src: log4j.properties
+    dest: "{{ spark_main_dir }}/conf/spark.properties"
+
+- name: Ensure Spark environmental variables are set
+  template:
+    src: spark-env.sh.j2
+    dest: /etc/profile.d/spark.sh
+    mode: 0644
+
+- name: reload profile
```

Comment:
Remove this task, as Ansible commands existing within a separate session, and the next command will already load anything in /etc/profile.d